### DISCUSSION TITLE
publishing from another file

### DISCUSSION THREAD

**Discussion by rushikeshkoli03** [uNetworking/uWebSockets.js#773] [Q&A]
I am running websocket server in app.js but for some reason I need the access to the app instance in another file to publish messages.
Is this possible?
So basically app.js run the websocket server and script.js is running a script and it generates some data every x seconds and I want to publish the outputs of this script to the connected clients in app.js.

**Comment by e3dio** [comment]
Node can open multiple script files with `import`

**Comment by e3dio** [comment]
If you are talking about 2 different Node processes you need Inter Process Communication with Net module https://nodejs.org/api/net.html#net

Or use Child Process https://nodejs.org/api/child_process.html

**Comment by rushikeshkoli03** [comment]
@e3dio  Thanks for the response. I will look into net module. I was also going through https://github.com/tardis-dev/serum-vial project and here they are using worker and broadcast channel for communication. Here is how i resolve
```
/* Simple pub/sub example (WIP) */

const uWS = require("uWebSockets.js");
const { BroadcastChannel } = require("worker_threads");
const abc = require("./abc");
const port = 9001;

let channel = new BroadcastChannel("testChannel", {
  type: "node",
  webWorkerSupport: true,
});

const app = (module.exports = uWS
  .App()
  .ws("/ws/conn", {
    open: (ws) => {
      ws.subscribe("root");
      app.publish("root", "g");
    },
    message: (ws, message, isBinary) => {
      app.publish("root", "gg");
    },
    drain: (ws) => {},
    close: (ws, code, message) => {},
  })
  .any("/ws/health", (res, req) => {
    res.end("Nothing to see here!");
  })
  .listen(port, (token) => {
    if (token) {
      console.log("Listening to port " + port);
    } else {
      console.log("Failed to listen to port " + port);
    }
    channel.onmessage = (message) => {
      console.log(message.data);
      app.publish("root", message.data);
    };
  }));

```
and the abc.js
```
const { BroadcastChannel } = require("worker_threads");

let channel = new BroadcastChannel("testChannel", {
  type: "node",
  webWorkerSupport: true,
});
setInterval(() => {
  channel.postMessage("hey");
}, 1000);

```
